src/emailpro/external-contact/add/emailpro-external-contact-add.controller.js

```diff
@@ -0,0 +1,110 @@
+        }
+
+        retrievingAlreadyTakenEmails () {
+            this.isLoading = true;
+
+            return this.EmailProExternalContacts
+                .retrievingContacts(this.$stateParams.productId)
+                .then((data) => {
+                    this.alreadyTakenEmails = data.list.results.map((contact) => contact.externalEmailAddress);
+                })
+                .catch((error) => {
+                    this.Alerter.alertFromSWS(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_fail"), error, this.$scope.alerts.dashboard);
+                    this.$scope.resetAction();
+                })
+                .finally(() => {
+                    this.isLoading = false;
+                });
+        }
+
+        submitting () {
+            this.$scope.resetAction();
+
+            if (!_.isEmpty(this.attachOrganization2010)) {
+                this.model.organization2010 = this.attachOrganization2010.name;
+            }
+
+            return this.EmailProExternalContacts
+                .addingContact(this.$stateParams.productId, this.model)
+                .then(() => {
+                    this.Alerter.success(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_success"), this.$scope.alerts.dashboard);
+                })
+                .catch((failure) => {
+                    this.Alerter.alertFromSWS(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_fail"), failure, this.$scope.alerts.dashboard);
+                });
+        }
+
+        isCurrentEmailAlreadyTaken () {
+            return this.alreadyTakenEmails.includes(this.model.externalEmailAddress);
+        }
+
+        getEmailErrorMessage () {
+            this.externalContactAddForm.emailAddress.$setValidity("email", true);
+
+            if (this.externalContactAddForm.emailAddress.$dirty && this.externalContactAddForm.emailAddress.$error.required) {
+                return this.translator.tr("emailpro_externalContacts_add_displayName_errors_isEmpty");
+            }
+
+            if (this.externalContactAddForm.emailAddress.$dirty && this.externalContactAddForm.emailAddress.$error.pattern) {
+                return this.translator.tr("emailpro_externalContacts_add_displayName_errors_notValid");
+            }
+
+            if (this.externalContactAddForm.emailAddress.$dirty && (this.isCurrentEmailAlreadyTaken() || this.externalContactAddForm.emailAddress.$error.email)) {
+                this.externalContactAddForm.emailAddress.$setValidity("email", false);
+                return this.translator.tr("emailpro_externalContacts_add_displayName_errors_alreadyTaken");
+            }
+
+            if (this.externalContactAddForm.emailAddress.$dirty && this.externalContactAddForm.emailAddress.$invalid) {
+                return this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_step1_email_invalid");
+            }
+
+            return "";
```

Comment:
Empty string :/, we should use `null` imo

Reply:
I don't think it mattered any way as it was used in a `ng-bind` but I removed the function in favor of `ng-messages`